target_finder/src/nodes/target_locator.cpp

```diff
@@ -43,92 +55,116 @@ using industrial_extrinsic_cal::ROSCameraObserver;
 using industrial_extrinsic_cal::Roi;
 using industrial_extrinsic_cal::Pose6d;
 using industrial_extrinsic_cal::Point3d;
+using industrial_extrinsic_cal::ROSListenerTransInterface;
 using target_finder::target_locator;
+using target_finder::target_verify;
+using target_finder::target_save_location;
+
 class TargetLocatorService
 {
 public:
   TargetLocatorService(ros::NodeHandle nh);
   ~TargetLocatorService(){};
   bool executeCallBack(target_locator::Request& req, target_locator::Response& res);
+  bool verifyCallBack(target_verify::Request& req, target_verify::Response& res);
+  bool saveLocCallBack(target_save_location::Request& req, target_save_location::Response& res);
   void initMCircleTarget(int rows, int cols, double circle_dia, double spacing);
-
+  void dynReConfCallBack(target_finder::target_finderConfig& config, uint32_t level);
+  Pose6d loadPose(std::string filepath);
+  
 private:
   ros::NodeHandle nh_;
-  ros::ServiceServer target_locate_server_;
+  ros::ServiceServer target_locate_server_; // provides the location of the target as a service
+  ros::ServiceServer target_verify_server_; // verifies the location matches stored location
+  ros::ServiceServer target_savelo_server_; // saves the location of the target to a file
   shared_ptr<Target> target_;
   string image_topic_;
   string camera_name_;
+  string camera_frame_;
+  string target_frame_;
+  string data_directory_;
   int target_type_;
   int target_rows_;
   int target_cols_;
+  boost::shared_ptr<dynamic_reconfigure::Server<target_finder::target_finderConfig> > reconf_srv_;
+  dynamic_reconfigure::Server<target_finder::target_finderConfig>::CallbackType reconf_CB_;
   ROSCameraObserver *camera_observer_;
+  ROSListenerTransInterface *target_to_camera_TI_;
+  // listen:pullTransform(), read:loadPose(scene_, file) and write:saveCurrentPose(scene, file)
+  // scene is optional if file is not ""
+  // otherwise scene is used to create the name of the file
+  // full file path takes the form data_directory_/filexxx.yaml
 };
 
 TargetLocatorService::TargetLocatorService(ros::NodeHandle nh)
 {
   nh_ = nh;
   ros::NodeHandle pnh("~");
 
-  int rows, cols;
-  double diameter, spacing;
+  // In launch you may also set the dynamic reconfigure variables of:
+  // target_locator/target_rows
+  // target_locator/target_cols
+  // target_locator/target_circle_dia
+  // target_locator/target_spacing
+
   if (!pnh.getParam("image_topic", image_topic_))
   {
     ROS_ERROR("Must set param:  image_topic");
   }
 
-  if (!pnh.getParam("target_type", target_type_))
-  {
-    ROS_ERROR("Must set param: target_type");
-  }
-
-  if (!pnh.getParam("camera_name", camera_name_))
+  if (!pnh.getParam("camera_name", camera_name_)) // need this to get the intrinsics 
   {
     ROS_ERROR("Must set param: camera_name");
   }
 
   camera_observer_ = new ROSCameraObserver(image_topic_, camera_name_);
```

Comment:
This pointer should be `delete`d in the destructor.  Otherwise, ROSCameraObserver's destructor will not be called when the pointer goes out of scope.  Most OSes will take back all memory when execution ends, but...

Reply:
delete's added